.github/workflows/ci.yml

```
@@ -26,42 +22,49 @@ on:
           - debug
 
 jobs:
-  test:
+  build_and_test:
     runs-on: ubuntu-latest
-    name: Run unit and integration tests
-    strategy:
-      matrix:
-        version: [20, 22]
+    name: Build and test
     steps:
-      - uses: actions/checkout@v4
-      - name: Using Node.js ${{ matrix.version }}
-        uses: actions/setup-node@v4
-        with:
-          node-version: ${{ matrix.version }}
-      - name: Install dependencies
-        run: npm install
-      - name: Run tests
-        run: npm test
-
-  build_and_publish:
-    needs: [test]
-    runs-on: ubuntu-latest
-    if: startsWith(github.ref, 'refs/tags')
-    steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@v5
 
-      - uses: actions/setup-node@v4
+      - uses: actions/setup-node@v5
         with:
           node-version: 22
-          registry-url: "https://registry.npmjs.org"
 
       - name: Install dependencies
-        run: npm install
+        run: npm ci
 
       - name: Build
         run: npm run build
 
-      - name: Publish
-        run: npm publish
-        env:
-          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+      - name: Test
+        run: npm test
+
+      - name: Upload dist artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: dist
+          path: dist/
+
+  publish:
+    needs: [build_and_test]
+    if: startsWith(github.ref, 'refs/tags/')
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+    steps:
+      - uses: actions/checkout@v5
+      - uses: actions/setup-node@v5
+        with:
+          node-version: "22"
+
+      - name: Download build artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: dist
+          path: dist/
+
+      - name: Publish (trusted publishing)
+        run: npm publish
```

I'm not familiar with this approach, but in theory it should work.

How about doing some tests? E.g. we can try something like this:

```suggestion
      - name: Pack
        run: npm pack
        
      - name: Upload package
        uses: actions/upload-artifact@v4
        with:
          name: package-tarball
          path: '*.tgz'
          retention-days: 7

      - name: Publish (trusted publishing)
        run: npm publish --dry-run
```